### PR TITLE
feat(bazel): Add clean_build option to synthtool

### PR DESCRIPTION
**Pull request by miraleung**
In `synth.py`, users will be able to enable build cleaning as follows. This is useful because there is otherwise no way to clear the Bazel cache via the shell.

This should temporarily affect only the PHP microgenerator. Its Bazel rules are implemented in such a way that clobbering the cache is needed if generator changes are made locally, but this will eventually be fixed in the microgenerator. Remote runs will not be affected.
```py
import synthtool.gcp as gcp

gapic = gcp.GAPICBazel()
gapic._clean_build = True
```

### CODE REVIEW

File: synthtool/gcp/gapic_bazel.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ def __init__(self):
         self._googleapis_private = None
         self._googleapis_discovery = None
         self._discovery_artifact_manager = None
+        self._clean_build = False
```
Comment (chingor13) [inline synthtool/gcp/gapic_bazel.py:50]: Probably prefer this to be set via the constructor args (or keyword arg to `_generate_code()`). `_variables` are meant to signal private so you shouldn't configure this via `gapic._clean_build = True`.

**Review by chingor13 — CHANGES_REQUESTED** [review]

**Comment by chingor13** [comment]
If the `clean_build` option isn't meant to be left in `synth.py`, consider making it a utility method on GAPICBazel to explicitly clear the cache (rather than as a step in `_generate_code()`).

File: synthtool/gcp/gapic_bazel.py (hunk 1 above)
Comment (miraleung) [reply synthtool/gcp/gapic_bazel.py]: Done.

**Comment by SurferJeffAtGoogle** [comment]
The code looks fine but I'm always hesitant to make changes that affect all languages.

In the synth.py files where we need to clean, could we instead call:
```.py
shell.run(["bazel", "clean", "--expunge", "--async"])
gapic = gcp.GAPICBazel()
gapic.php_library(..)
```

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]
See prior comment.

**Comment by miraleung** [comment]
> The code looks fine but I'm always hesitant to make changes that affect all languages.
> 
> In the synth.py files where we need to clean, could we instead call:
> 
> ```python
> shell.run(["bazel", "clean", "--expunge", "--async"])
> gapic = gcp.GAPICBazel()
> gapic.php_library(..)
> ```

I'd tried this, but this throws an error since it's not executed in the Bazel workspace (which is accessible only inside `gcp.GAPICBazel()`(.

File: synthtool/gcp/gapic_bazel.py [hunk 2]
```diff
@@ -182,6 +189,10 @@ def _generate_code(
         cwd = os.getcwd()
         os.chdir(str(api_definitions_repo))
 
+        if clean_build:
```
Comment (SurferJeffAtGoogle) [inline synthtool/gcp/gapic_bazel.py:192]: I'd love to refactor it `_generate_code()`, turn it inside out, so there are 7 top-level functions calling common functions as needed, with no if statements.  But, I'm afraid of breaking everything.

Sorry you had to deal with this, Mira.

**Review by SurferJeffAtGoogle — APPROVED** [review]

**Comment by SurferJeffAtGoogle** [comment]
I'll approve as soon as Kokoro tests pass.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**